Fixed Aquamacs command-line tool

HayoBaan opened the pull request:
With --args, the aquamacs command no longer was able to open files from e.g. the current directory.

HayoBaan committed:
Fixed Aquamacs command-line tool.

You can now specify files and arguments (mostly) as expected.

davidswelt commented:
I confirm the bug.  This change, however, now breaks giving Aquamacs additional arguments, which is why this was there in the first place.  Would it be possible to check whether strings start with -- and if they don't, give them as filenames?
Also, suggest using a description of the change, not of the bug.

HayoBaan commented:
Hi David,

You are right, it breaks providing “normal” args now. But I think the behaviour now is MUCH worse: you now no longer are able to open any files specified on relative paths (which, I think, is what most of us do).

Hmm, I just noticed another fundamental flaw in the script: in the current solution it would create temp files for all options! E.g. `aquamacs -q` would create a file called `-q`.

So how about this: we scan the arguments and replace the files with their absolute path and leave all things that look like options (e.g. starting with `-` and that are not already files) alone?

HayoBaan commented:
Been playing a bit with this and found other issues with the --args approach:
1. When you start aquamacs for the first time while specifying multiple files (using open --args), files will not open in tabs like they used to do, but you get a vertically split fram (with the bottom showing a buffer list if more than 2 files where specified) 
2. When aquamacs is already open, additional calls of the aquamacs script (using open --args) will NOT open the additional files.

So, for files at least, --args is NOT a good solution (the only advantage is that you then don't need temp files, but that is moot if a second run of the script doesn't open them).

HayoBaan commented:
Hi David, I think I solved this as best as possible now:

* Temp files are only created for non-existing files that don't start with `-`.
* Files are placed as argument to open just like normal.
* Arguments (options) starting with `-` are placed after `--args` in the `open` command.

I also cleaned-up the script a bit.
Please note that I rebased and force pushed the updates I made so you can more easily merge them in the master aquamacs3 branch.

davidswelt reviewed (approved):
Excellent change.  Thank you for working on this.
